Log zigbee-herdsman version on startup

zombielinux opened the pull request:
Allow current zigbee-herdsman version to be exposed to controller for logging and to startup output.

zombielinux committed:
add getZigbeeHerdsmanVersion Function

Allow current zigbee-herdsman version to be exposed to controller for logging and to startup output.

zombielinux committed:
Update utils.test.js

Add tests for getZigbeeHerdsmanVersion() function

zombielinux committed:
Merge pull request #1 from zombielinux/patch-2

Update utils.test.js

zombielinux committed:
Update utils.js

Added getZigbeeHerdsman to module exports

zombielinux committed:
Update utils.js

Fixed path because I'm dumb.

zombielinux committed:
Update utils.test.js

Apparently version numbers are important when they're different.

zombielinux committed:
Update zigbee.js

Added actual display of herdsman version output.

zombielinux commented:
Mercy that took a lot. Don't code with toddlers is all I have to say. Sorry for the mess.

nurikk commented on lib/util/utils.js:
```diff
@@ -69,6 +69,30 @@ async function getZigbee2mqttVersion() {
     });
 }
 
```
Looks very similar to getZigbee2mqttVersion, can it be generalized to reduce code bloating?

zombielinux replied:
I'm sure it could, or to be extended to take an argument for any included dependent packages. 

I figured we could get started with this since "starting zigbee-herdsman" is currently displayed on startup, and none of the others are.

zombielinux replied:
@nurikk I ended up just creating a separate generalized version lookup function, trimmed a lot of fat down, and extensible as dependencies are added and removed. 

Thanks for the inspiration!

nurikk replied:
Good job 👍

Koenkk commented on lib/util/utils.js:
```diff
@@ -69,6 +69,30 @@ async function getZigbee2mqttVersion() {
+
+            if (err) {
+                try {
+                    commitHash = require('../../node_modules/zigbee-herdsman/.hash.json').hash;
```
I think the git commit part is overkill here and also will not work in the Docker installations (because it's not a git repo there and there is no `.hash.json`). Relying on the zigbee-herdsman version in package.json is good enough.

zombielinux replied:
@Koenkk  I saw that too. I removed the git dependency, created a new test, and generalized the function so it could be reused on any dependency in node_modules.

zombielinux committed:
Update utils.js

Remove git checking in getZigbeeHerdsmanVersion()

zombielinux committed:
Merge branch 'dev' into patch-1

zombielinux committed:
Update utils.test.js

Remove commitHash from getZigbeeHerdsmanVersion test

zombielinux committed:
Update utils.js

Remove commitHash from getZigbeeHerdsmanVersion

zombielinux committed:
Update utils.js

zombielinux committed:
Update utils.test.js

zombielinux committed:
Update utils.js

Try removing less code this time.

zombielinux committed:
Update utils.test.js

zombielinux committed:
Update zigbee.js

zombielinux committed:
Update utils.js

zombielinux committed:
Update utils.js

zombielinux committed:
Update utils.test.js

zombielinux committed:
Update utils.js

zombielinux committed:
Update utils.test.js

zombielinux committed:
Update utils.js

zombielinux committed:
Update utils.js

zombielinux committed:
Update zigbee.js

zombielinux committed:
Update utils.js

Generalize getZigbeeHerdsmanVersion() to getDependVersion(), so any node_module dependency can be checked

zombielinux committed:
Update zigbee.js

zombielinux committed:
Update utils.test.js

zombielinux committed:
Update utils.js

zombielinux committed:
Update zigbee.js

zombielinux committed:
Update utils.test.js

zombielinux committed:
Update zigbee.js

zombielinux committed:
Update zigbee.js

zombielinux committed:
Update zigbee.js

zombielinux committed:
Update zigbee.js

zombielinux committed:
Update zigbee.js

zombielinux commented:
Alright, one generalized dependency version lookup function, currently implemented to print the zigbee-herdsman version out and tested against the same dependency.

zombielinux committed:
Merge branch 'dev' into patch-1

zombielinux committed:
Merge branch 'dev' into patch-1

Koenkk committed:
Update utils.js

Koenkk committed:
Update zigbee.js

Koenkk committed:
Update utils.test.js

Koenkk committed:
Merge branch 'dev' into patch-1

Koenkk committed:
Update zigbee.js

Koenkk commented:
Thanks!
